asyncpg/types.py

```diff
@@ -85,6 +85,52 @@ def upper_inf(self):
     def isempty(self):
         return self._empty
 
+    def issubset(self, other):
+        if self._empty:
+            return True
+        if other._empty:
+            return False
+
+        if other._lower is None:
+            lowerOk = True
+        else:
+            if self._lower is None:
+                return False
+            if other._lower_inc or not self._lower_inc:
+                lowerOk = self._lower >= other._lower
+            else:
+                lowerOk = self._lower > other._lower
+
+        if not lowerOk:
+            return False
+
+        if other._upper is None:
+            upperOk = True
+        else:
+            if self._upper is None:
+                return False
+            if other._upper_inc or not self._upper_inc:
+                upperOk = self._upper <= other._upper
+            else:
+                upperOk = self._upper < other._upper
+
+        return lowerOk and upperOk
+
+    def issuperset(self, other):
+        return other.issubset(self)
+
+    def size(self, step=None):
```

Comment:
I do not understand why the *step* argument is necessary.  The size of the range is determined unambiguously by whether the upper and lower bounds are inclusive.

Reply:
The returned size for [1,2) and [1,2] should be different. In postgres they have a distinction between a discrete and continuous range. If this was a discrete integer range the results would be `2-1 = 1` and `2-1+1=2`. `+1` since the discrete step for an integer is `1`. What is the step value for  continuous range like a float. Or if using datetime what is your lowest time increment you care about.

Reply:
OK, I think the generic `Range` class shouldn't be implementing the `size()` method then.  It's too dependent on the data type and, since Postgres allows user-defined range types, hardcoding it for int, float, and date types wouldn't save us.

Reply:
The distinction between "discrete" and "continuous" ranges has an impact also of `issubset()` and `issuperset()`.

Consider the following two ranges: (1,2]::int4range and [2,3)::int4range. They represent the very same set of values, and indeed `select '(1,2]'::int4range <@ '[2,3)'::int4range` returns True, but

```python
>>> r1 = asyncpg.Range(1, 2, lower_inc=False, upper_inc=True)
>>> r2 = asyncpg.Range(2, 3, lower_inc=True, upper_inc=False)
>>> r1.issubset(r2)
False
```